### PR TITLE
feat: Support for contract addresses in test and main net

### PR DESCRIPTION
**Pull request by saltPink**

### CODE REVIEW

File: src/config/constants/nfts.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
 import { Nft } from './types'
 
-export const RABBIT_MINTING_FARM_ADDRESS = '0x7c8b60d2b859a38c8B9b5B6CB4565485cb637c7a'
-export const PANCAKE_RABBITS_ADDRESS = '0xDf7952B35f24aCF7fC0487D01c8d5690a60DBa07'
+export const RABBIT_MINTING_FARM_ADDRESS = {
```
Comment (saltKosher) [inline src/config/constants/nfts.ts:3]: You've missed `RABBIT_MINTING_FARM_ADDRESS[CHAIN_ID]` in these files:

`src/views/Nft/contexts/NftProvider.tsx`

`src/views/Nft/components/NftGlobalNotification.tsx`

Comment (jonasman) [reply src/config/constants/nfts.ts:3]: where? i think it is correct

Comment (jonasman) [reply src/config/constants/nfts.ts:3]: where? i think it is correct

File: src/config/constants/pools.ts [hunk 2]
```diff
@@ -5,7 +5,31 @@ const pools: PoolConfig[] = [
     sousId: 0,
     tokenName: 'CAKE',
     stakingTokenName: QuoteToken.SALT,
-    stakingTokenAddress: '0x2849b1aE7E04A3D9Bc288673A92477CF63F28aF4', // SALT
+    stakingTokenAddress: {
```
Comment (saltKosher) [inline src/config/constants/pools.ts:8]: Missed `src/state/pools/fetchPools.ts`

`stakingTokenAddress[CHAIN_ID]`

File: src/hooks/useContract.ts [hunk 3]
```diff
@@ -47,7 +49,7 @@ export const useERC20 = (address: string) => {
   return useContract(erc20Abi, address)
 }
 
-export const useCake = () => useERC20(getCakeAddress())
+export const useCake = () => useERC20(getCakeAddress()[CHAIN_ID])
```
Comment (saltKosher) [inline src/hooks/useContract.ts:52]: Not like this, go inside the `getCakeAddress()` and add it there.

`export const getCakeAddress = () => addresses.cake[CHAIN_ID]`

File: src/hooks/useContract.ts [hunk 4]
```diff
@@ -61,22 +63,22 @@ export const usePancakeRabbits = (address: string) => {
 
 export const useLottery = () => {
   const abi = (lottery as unknown) as AbiItem
-  return useContract(abi, getLotteryAddress())
+  return useContract(abi, getLotteryAddress()[CHAIN_ID])
```
Comment (saltKosher) [inline src/hooks/useContract.ts:66]: Same for these ones...

File: src/config/constants/pools.ts (hunk 2 above)
Comment (saltPink) [reply src/config/constants/pools.ts:8]: ` src/state/pools/fetchPools.ts` is all converted

File: src/hooks/useTokenBalance.ts [hunk 5]
```diff
@@ -33,7 +35,7 @@ export const useTotalSupply = () => {
 
   useEffect(() => {
     async function fetchTotalSupply() {
-      const cakeContract = getContract(cakeABI, getCakeAddress())
+      const cakeContract = getContract(cakeABI, getCakeAddress()[CHAIN_ID])
```
Comment (saltKosher) [inline src/hooks/useTokenBalance.ts:38]: Same here, go into `getCakeAddress()`

**Pull request closed without merging**